pkg/skaffold/build/buildpacks/lifecycle.go

```diff
@@ -48,17 +50,25 @@ func (b *Builder) build(ctx context.Context, out io.Writer, a *latest.Artifact,
 
 	builderImage := artifact.Builder
 	logrus.Debugln("Builder image", builderImage)
-	if err := b.pull(ctx, out, builderImage, artifact.ForcePull); err != nil {
-		return "", err
+	if !artifact.ForcePull {
+		if err := b.pull(ctx, out, builderImage); err != nil {
+			return "", err
```

Comment:
This is confusing: if we aren't force-pulling then we pull the image?  Doesn't pack do this anyways?  Or this needs a comment to explain why.

Reply:
pack is either pulling always or pulling never. This is trying to pullIfNotPresent. I'll update the code with a comment.